explorer/scan/local_scanner.go

```diff
@@ -302,6 +302,7 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 
 	justAssets := []*inventory.Asset{}
 	for _, asset := range assets {
+		asset.asset = asset.runtime.Provider.Connection.Asset // to ensure we get all the information the connect call gave us
```

Comment:
i think we should set this right after the `Connect` for that asset returns, such that any guards/checks that follow are executed against just one asset